service/matching/tasklist/task_list_manager_test.go

```diff
@@ -532,6 +533,186 @@ func TestMisconfiguredZoneDoesNotBlock(t *testing.T) {
 	assert.True(t, breakRetryLoop, "task should be dispatched to default channel")
 }
 
+func TestGetIsolationGroupForTask(t *testing.T) {
+	defaultAvailableIsolationGroups := []string{
+		"a", "b", "c",
+	}
+	taskIsolationPollerWindow := time.Second * 10
+	testCases := []struct {
+		name                     string
+		taskIsolationGroup       string
+		taskIsolationDuration    time.Duration
+		taskLatency              time.Duration
+		availableIsolationGroups []string
+		recentPollers            []string
+		expectedGroup            string
+		expectedDuration         time.Duration
+		// If false (the default) we don't advance time before calling the function. The function behaves differently
+		// during the initial taskIsolationPollerWindow
+		justStarted          bool
+		disableTaskIsolation bool
+	}{
+		{
+			name:                     "success - startup allows any group",
+			taskIsolationGroup:       "a",
+			taskIsolationDuration:    time.Second,
+			availableIsolationGroups: defaultAvailableIsolationGroups,
+			expectedGroup:            "a",
+			expectedDuration:         time.Second,
+			recentPollers:            []string{"b"},
+			justStarted:              true,
+		},
+		{
+			name:                     "success - no recent pollers allows any group",
+			taskIsolationGroup:       "a",
+			taskIsolationDuration:    time.Second,
+			availableIsolationGroups: defaultAvailableIsolationGroups,
+			expectedGroup:            "a",
+			expectedDuration:         time.Second,
+		},
+		{
+			name:                     "success - recent poller allows group",
+			taskIsolationGroup:       "a",
+			taskIsolationDuration:    time.Second,
+			availableIsolationGroups: defaultAvailableIsolationGroups,
+			expectedGroup:            "a",
+			expectedDuration:         time.Second,
+			recentPollers:            []string{"a"},
+		},
+		{
+			name:                     "success - low task latency",
+			taskIsolationGroup:       "a",
+			taskIsolationDuration:    time.Second,
+			taskLatency:              time.Millisecond * 500,
+			availableIsolationGroups: defaultAvailableIsolationGroups,
+			expectedGroup:            "a",
+			expectedDuration:         time.Millisecond * 500,
```

Comment:
expectedDuration is 1s - 0.5s but it confused me at first because expectedDuration == taskLatency.
let's change taskLatencies from 0.5s to 0.3s so corresponding duration becomes 0.7s to avoid confusion